editor/components/page-attributes/template.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { isEmpty, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { withContext, withInstanceId } from '@wordpress/components';
+import { compose } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { getEditedPostAttribute } from '../../store/selectors';
+import { editPost } from '../../store/actions';
+
+export function PageTemplate( { availableTemplates, selectedTemplate, instanceId, onUpdate } ) {
+	if ( isEmpty( availableTemplates ) ) {
+		return null;
+	}
+	const selectId = `template-selector-${ instanceId }`;
+	const onEventUpdate = ( event ) => onUpdate( event.target.value );
+	return (
+		<div className="editor-page-attributes__template">
+			<label htmlFor={ selectId }>{ __( 'Template:' ) }</label>
+			<select
+				id={ selectId }
+				value={ selectedTemplate }
+				onBlur={ onEventUpdate }
+				onChange={ onEventUpdate }
+			>
+				{ map( { '': __( 'Default template' ), ...availableTemplates }, ( templateName, templateSlug ) => (
+					<option key={ templateSlug } value={ templateSlug }>{ templateName }</option>
+				) ) }
+			</select>
+		</div>
+	);
+}
+
+const applyConnect = connect(
+	( state ) => {
+		return {
+			selectedTemplate: getEditedPostAttribute( state, 'template' ),
+		};
+	},
+	{
+		onUpdate( templateSlug ) {
+			return editPost( { template: templateSlug || '' } );
+		},
+	}
+);
+
+const applyWithContext = withContext( 'editor' )(
+	( settings ) => ( {
+		availableTemplates: settings.availableTemplates,
+	} )
+);
+
+export default compose( [
```

Comment:
You don't need to use an array with `compose`:

```js
export default compose(
	applyConnect,
	applyWithContext,
	withInstanceId
)( PageTemplate );```

Is fine, too :)

Reply:
I always go back and forth on this one. I was going to say that one upside of passing an array is that all functions in it can be formatted alike, notably with a trailing comma, **but** it seems that with our current linting + build rules the following is accepted:

```js
export default compose(
  applyFoo,
  applyBar, // notice trailing comma
)( FooBar );
```

TIL :)

Reply:
Right, you can now add trailing comma when using ES.next 👍